iree/compiler/Conversion/Common/LinalgBufferizePass.cpp

```diff
@@ -541,80 +604,338 @@ LogicalResult preProcessInterfaceStoreTensorOp(
+                 offsets, sizes, strides)
+                 .cast<MemRefType>()
+           : MemRefType());
+  return b.create<memref::SubViewOp>(loc, subViewResultType, inputBuffer,
+                                     offsets, sizes, strides);
+}
+
+/// Computes the `memrefs` to use for the result of an operation based on
+/// - If the result has a tied operand reuse the buffer for the tied operand (or
+///   an alias of it) as the buffer for the result. The `tiedOperands` vector is
+///   expected to be as large as the number of results.
+/// - If the result has no tied operands, the corresponding position in the
+///   `tiedOperands` list must be `nullptr`.
+/// - If the result is in the same equivalence set as the result of the dispatch
+///   region (i.e. `value` operand of a `flow.dispatch.tensor.store`) then
+///   return an alias/view of the buffer passed into the dispatch region to
+///   store the results.
+/// - Lastly, allocate a temporary buffer for the result using the passed
+///   allocation function.
+static LogicalResult getOrAllocateResultBuffers(
+    OpBuilder &b, Operation *op, ArrayRef<Value> tiedOperands,
+    BlockAndValueMapping &bvm, BufferizationPlan &plan,
+    WorkgroupMemoryAllocationFn allocationFn) {
+  for (auto result : llvm::enumerate(op->getResults())) {
+    if (bvm.lookupOrNull(result.value())) continue;
+    Value buffer;
+    if (tiedOperands[result.index()] &&
+        plan.isEquivalent(tiedOperands[result.index()], result.value())) {
+      buffer =
+          TypeSwitch<Operation *, Value>(op)
+              .Case<linalg::TensorReshapeOp>(
+                  [&](linalg::TensorReshapeOp reshapeOp) {
+                    return getAliasingBufferForReshapeResult(b, reshapeOp, bvm);
+                  })
+              .Case<SubTensorInsertOp>(
+                  [&](SubTensorInsertOp subTensorInsertOp) {
+                    return bvm.lookupOrNull(subTensorInsertOp.dest());
+                  })
+              .Case<SubTensorOp>([&](SubTensorOp subTensorOp) {
+                return getAliasingBufferForSubtensorResult(b, subTensorOp, bvm);
+              })
+              .Case<tensor::CastOp>([&](tensor::CastOp castOp) {
+                return getAliasingBufferForCastResult(b, castOp, bvm);
+              })
+              .Case<linalg::LinalgOp>([&](linalg::LinalgOp linalgOp) {
+                return bvm.lookupOrNull(linalgOp.getOutput(result.index()));
+              })
+              .Default([&](Operation *op) { return nullptr; });
+    }
+    if (!buffer && plan.isInStoreSet(result.value())) {
+      buffer = getInplaceResultBuffer(b, result.value(), bvm);
+    }
+    if (!buffer) {
+      buffer = allocateBufferForResult(b, op, allocationFn);
+    }
+    if (!buffer) {
+      return op->emitError("unable to get result buffer for op");
+    }
+    bvm.map(result.value(), buffer);
+    DEBUG_WITH_TYPE(DEBUG_TYPE, {
```

Comment:
I see multiple such debug snippets with the same structure. Adding some leading description will make it clear what each of them is for? (I'm basically thinking I might need to read this some day later. ;)

Reply:
They are just printing which memref value is created for a tensor value. I am not sure what to add there. I'll just leave it as is, and we can add to that as needed.